sql_joins_demystified/sql_joins_demystified.md

```diff
@@ -0,0 +1,589 @@
+
+**One Table?  Not a Good Idea!**
+
+In short, if we try to force all the data we care about into a single table, we have a problem of rapidly multiplying columns and data repetition in rows.  This makes maintaining, cleaning, correcting, and understanding data very tricky.
+
+We're much better off collecting various groups of facts about our research subjects into distinct, separate tables, like a "demographics" table, a "depression" table, an "anxiety" table, an "address" table, and so on, including a "subj\_id" column to link rows back to the subject in question.  This realization, of how we can separate data into concepts that each get a table, leads us to the idea of **normalization**.
+
+## Normalization
+
+When data is normalized, we organize data to reduce the possibility of needless data duplication and empty cells.  Normalization is a complex information science topic and has different degrees of completeness, so here we'll talk about what normalization **looks like** more than what it **is**.  
+
+<div class = "behind-the-scenes">
+<b style="color: rgb(var(--color-highlight));">Behind the scenes</b><br>
+In this module, we're discussing a kind of normalization called "third normal form", often abbreviated 3NF.  The theory behind normalization has its own challenging jargon, like "cardinality", "atomicity", "decomposition", etc.  If you get excited by database optimization and information science, you might enjoy a deep dive into normalization.  But for most of us, we only need to develop a casual understanding of how normalization affects our databases. 
+</div>
+
+**Entities: What Kinds of Things Exist?**
+
+In a normalized database, we separate data into tables representing different logical **entities**, like "address", "procedure", "medication", "device", etc., and give each of these entities an identifier (like "device\_id" or "medication\_id") to identify a particular instance of the entity.  
+
+This id number can also be a **primary key** if it will only appear once in the entity table (say, for an item identifier that you don't want to be duplicated).  Imagine a "medication" table with rows that include a medication id and information about that medication (like its brand name, generic name, and whether it requires a prescription).  Or think of an "item" table which includes an item id, along with details about that item like the manufacturer and description.  Each row is unique, and the id as a primary key won't repeat (SQL enforces this and won't let you add a new row with the same primary key).
```

Comment:
```suggestion
This id number can also be a **primary key** if it will only appear once in the entity table (say, for an item identifier that you don't want to be duplicated).  Imagine a "medication" table with rows that include a medication ID and information about that medication (like its brand name, generic name, and whether it requires a prescription).  Or think of an "item" table which includes an item ID, along with details about that item like the manufacturer and description.  Each row is unique, and the id as a primary key won't repeat (SQL enforces this and won't let you add a new row with the same primary key).
```

There seems to be a lot of discussion on whether "ID" as short for "identifier" should be call capitalized or written "Id." as an abbreviation. My vote is for "ID"